test/decompress_SUITE.erl

```diff
@@ -0,0 +1,327 @@
+		[{<<"content-encoding">>, <<"gzip">>}], Body, Config),
+	{_, <<"identity, gzip">>} = lists:keyfind(<<"accept-encoding">>, 1, Headers),
+	ok.
+
+override_accept_encoding_header_invalid(Config) ->
+	doc("Header accept-encoding must override invalid accept-encoding."),
+	Data = <<"test">>,
+	Body = zlib:gzip(Data),
+	{200, Headers, Data} = do_post("/invalid-header",
+		[{<<"content-encoding">>, <<"gzip">>}], Body, Config),
+	{_, <<"gzip">>} = lists:keyfind(<<"accept-encoding">>, 1, Headers),
+	ok.
+
+%% RFC 9110. Section 12.5.3. 10.3. If the representation's content coding is
+%% one of the content codings listed in the Accept-Encoding field value, then
+%% it is acceptable unless it is accompanied by a qvalue of 0.
+%%
+%% gzip must not have a qvalue of 0 when the handler is used. Set to 1.
+override_accept_encoding_excluded(Config) ->
+	doc("Header accept-encoding must override when explicitly excluded."),
+	Data = <<"test">>,
+	Body = zlib:gzip(Data),
+	{200, Headers, Data} = do_post("/reject-explicit-header",
+		[{<<"content-encoding">>, <<"gzip">>}], Body, Config),
+	{_, <<"identity;q=1, gzip;q=1">>} = lists:keyfind(<<"accept-encoding">>, 1, Headers),
+	ok.
+
+%% RFC 9110. Section 12.5.3. 10.2. If the representation has no content coding,
+%% then it is acceptable by default unless specifically excluded by the
+%% Accept-Encoding field stating either "identity;q=0" or "*;q=0" wihout a more
+%% specific entry for "identity".
+%%
+%% *;q=0 will reject codings that are not listed. Specific entry gzip must
+%% always be listed when the handler is used. Add gzip.
+add_accept_encoding_excluded(Config) ->
+	doc("Header accept-encoding must added when implicitly excluded."),
+	Data = <<"test">>,
+	Body = zlib:gzip(Data),
+	{200, Headers, Data} = do_post("/reject-implicit-header",
+		[{<<"content-encoding">>, <<"gzip">>}], Body, Config),
+	{_, <<"gzip;q=1, identity;q=1, *;q=0">>} = lists:keyfind(<<"accept-encoding">>, 1, Headers),
+	ok.
+
+no_override_accept_coding_set_explicit(Config) ->
+	doc("Confirm that accept-encoding is not overridden when explicitly set."),
+	Data = <<"test">>,
+	Body = zlib:gzip(Data),
+	{200, Headers, Data} = do_post("/accept-explicit-header",
+		[{<<"content-encoding">>, <<"gzip">>}], Body, Config),
+	{_, <<"identity, gzip;q=0.5">>} = lists:keyfind(<<"accept-encoding">>, 1, Headers),
+	ok.
+
+no_override_accept_coding_set_implicit(Config) ->
+	doc("Confirm that accept-encoding is not overridden when implicitly set."),
+	Data = <<"test">>,
+	Body = zlib:gzip(Data),
+	{200, Headers, Data} = do_post("/accept-implicit-header",
+		[{<<"content-encoding">>, <<"gzip">>}], Body, Config),
+	{_, <<"identity, *;q=0.5">>} = lists:keyfind(<<"accept-encoding">>, 1, Headers),
+	ok.
+
+%% RFC 9110. Section 12.5.3. 10.1. If no Accept-Encoding header field is in the
+%% request, any content coding is considered acceptable by the user agent.
+%%
+%% Don't add anything on error or when that handler is not used.
```

Comment:
Just some info: this comment refers to how user agents are behaving, there is nothing to do here as far as Cowboy is concerned. It could only be relevant to the compression stream handler, not the decompression one.